app/models/item.js

```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+var Types = Schema.Types;
+var ObjectId = Types.ObjectId;
+
+var QueryPlugin = require('mongoose-query');
+
+var ItemSchema = new Schema({
+  barcode : { type: String, unique:true },
```

Comment:
After further investigation it seems this still creates an issue, but by adding sparse, we can effectively allow undefined barcodes and still make sure those that have been defined are unique.

"_An index that is both sparse and unique prevents collection from having documents with duplicate values for a field but allows multiple documents that omit the key._"
https://docs.mongodb.com/manual/core/index-sparse/